### ISSUE TITLE
Task 3: Image search using Pinecone

### ISSUE THREAD

**Issue by parvashah-create** [parvashah-create/image-search#3]
# Description
Task 3: Image search using Pinecone
You realize search is slow in Task 1 when you increase the images from 100 to 10000.
You are approached by Pinecode to use their vector database to speed up search.
They provide this ref example:
https://github.com/pinecone-io/examples/blob/master/search/image/image-search/i
mage-search.ipynb
Use this approach and integrate with Streamlit for the data used in Task 1. Compare
performance of Task 1 and Task 3

# Steps
1. Install the necessary libraries: You will need to install Streamlit, Pinecone, PyTorch, Torchvision, NumPy, and Pillow. You can install them using pip.

2. Download the DeepFashion dataset: Download the dataset from https://mmlab.ie.cuhk.edu.hk/projects/DeepFashion.html and extract it to a folder.

3. Download the pre-trained model: Download the pre-trained model from https://drive.google.com/file/d/1QFRs_MsT6T_3UNrSJdDfJbW8XZKZXU6b/view and extract it to a folder.

4. Load the model: Load the pre-trained model into your application using PyTorch.

5. Create a Pinecone index: Use Pinecone to create an index for the feature vectors of the images in the DeepFashion dataset.

6. Preprocess the images: Preprocess the images in the DeepFashion dataset by resizing them to the size required by the pre-trained model and normalizing them.

7. Encode the images: Use the pre-trained model to encode the images into feature vectors.

8. Add feature vectors to Pinecone index: Add the feature vectors to the Pinecone index.

9. Create a Streamlit interface: Use Streamlit to create a user interface where users can upload an image and search for similar images in the DeepFashion dataset.

10. Preprocess the uploaded image: Preprocess the uploaded image by resizing it to the size required by the pre-trained model and normalizing it.

11. Encode the image: Use the pre-trained model to encode the image into a feature vector.

12. Search for similar images: Use Pinecone to search the index for images that are similar to the uploaded image based on the encoded feature vector.

13. Display the results: Display the similar images found in the DeepFashion dataset to the user in the Streamlit interface.

14. Compare performance: Measure the performance of the Pinecone-based image search and compare it to the performance of the approach used in Task 1.

15. Deploy the application: Deploy the application to a server or platform where it can be accessed by users.
